src/views/Main/Catalog/Components/NameNamespaceAutosuggest.js

```diff
@@ -0,0 +1,161 @@
+import React, { PropTypes } from 'react';
+import Autosuggest from 'react-autosuggest';
+import { Button } from 'react-bootstrap';
+import '../../../../styles/autosuggest.css';
+
+const getSuggestionValue = suggestion => suggestion;
+
+const renderSuggestion = suggestion => (
+  <div>
+    {suggestion}
+  </div>
+);
+
+export class NameNamespaceAutosuggest extends React.Component {
+  static propTypes = {
+    id: PropTypes.number,
+    names: PropTypes.object,
+    namespaces: PropTypes.object,
+    className: PropTypes.string,
+    addProperty: PropTypes.func,
+    type: PropTypes.string
+  }
+
+  constructor() {
+    super();
+    this.state = {
+      nameVal: '',
+      nameSuggestions: [],
+      namespaceVal: '',
+      namespaceSuggestions: []
+    };
+  }
+
+  getNamespaceVal = () => document
+    .getElementById('newNamespace'
+      .concat(this.props.type)
+      .concat(this.props.id)
+    ).firstChild
+    .firstChild
+    .value;
+
+  getNameVal = () => document
+    .getElementById('newName'
+      .concat(this.props.type)
+      .concat(this.props.id)
+    ).firstChild
+    .firstChild
+    .value;
+
+  getSuggestions(focused, unfocused, focusedList, unfocusedList) {
```

Comment:
so if you remove it, you can change `onNameSuggestionsFetchRequested` and `onNamespaceSuggestionsFetchRequested` since you don't need to pass in the 3rd parameter anymore.

with that change, my followup question is, why do you need both `names` and `namespaces`? it seems like you're only ever interacting with one of the two